index.js

```diff
@@ -22,35 +16,31 @@ app.use(express.json());
 app.use(express.static("public"));
 
 io.on("connection", (socket) => {
-  const handleSignin = (args) => {
-    const { lat, lon } = args || {};
+  let { hash: userHash } = socket.handshake.query;
 
-    if (!lat || !lon) {
-      return;
-    }
-
-    const hash = geohash.encode(lat, lon, 5);
-
-    const randomName = uniqueNamesGenerator({
-      dictionaries: [adjectives, colors, animals],
-      separator: "-",
-    }); // big-red-donkey
-
-    let grid = grids[hash];
+  if (userHash) {
+    // Join the user to the grid
+    socket.join(userHash);
 
-    if (!grid) {
-      grids[hash] = new Set();
-      grid = grids[hash];
+    // Send all users in the grid to the newly connected user
```

Comment:
Is this comment accurate ? `io.to(userHash).emit("users", [...grids[userHash]]);` this is sending the user list to everyone in that room not just the newly connected user